### PR TITLE
swaps: fix conditional to use fast speed

### PR DESCRIPTION
**Pull request by estebanmino**
Fixes TEAM2-290
Figma link (if any):

## What changed (plus any additional context for devs)

this conditional was very wrong

```
    !gasParams.maxFeePerGas ||
    !gasParams.maxPriorityFeePerGas ||
    !gasParams.gasPrice
```

it will always be true since in mainnet `!gasParams.gasPrice` is true because of eip1559, and L2s `!gasParams.maxFeePerGas || !gasParams.maxPriorityFeePerGas ||` is also true because there are no eip1559 params, so we always use fast

i changed it to check for `!gasParams.gasPrice` on L2s and `!gasParams.maxFeePerGas || !gasParams.maxPriorityFeePerGas` for mainnet

## What to test

set custom gas and send a ETH <>. TOKEN swap, gas should be respected

## Final checklist

- [ ] Assigned individual reviewers?
- [ ] Added labels? (`team1/team2`, `critical path`, `release`, `dev QA`)
- [ ] Did you test both iOS and Android?
- [ ] If your changes are visual, did you check both the light and dark themes?
- [ ] Added e2e tests? If not, please specify why
- [ ] If you added new critical path files, did you update the CODEOWNERS file?
- [ ] If no `dev QA` label, did you add the PR to the QA Queue?

### CODE REVIEW

**Review by ibrahimtaveras00 — APPROVED** [review]
fix looks good, QA Passed 👍🏽

File: src/raps/actions/swap.ts [hunk 1]
```diff
@@ -47,12 +47,14 @@ const swap = async (
...
   let gasParams = parseGasParamsForTransaction(selectedGasFee);
   // if swap isn't the last action, use fast gas or custom (whatever is faster)
-  if (
-    currentRap.actions.length - 1 > index ||
-    !gasParams.maxFeePerGas ||
-    !gasParams.maxPriorityFeePerGas ||
-    !gasParams.gasPrice
-  ) {
+  const isL2 = isL2Network(
+    ethereumUtils.getNetworkFromChainId(parameters?.chainId || 1)
```
Comment (derHowie) [inline src/raps/actions/swap.ts:51]:
```
import { ChainId } from '@uniswap/sdk';
const isL2 = isL2Network( ethereumUtils.getNetworkFromChainId(parameters?.chainId || ChainId.MAINNET)
```

Nit picky suggestion, but if you ever need to see where we are using mainnet chain id defaults, this is a lot easier to search for than `1`.

Comment (estebanmino) [reply src/raps/actions/swap.ts:51]: oh yes nice, i'm gonna try to find a way to do the same without `@uniswap/sdk` since we don't have it anymore

Comment (estebanmino) [reply src/raps/actions/swap.ts:51]: ah is the same with `@rainbow-me/swaps` 🤝

Comment (estebanmino) [reply src/raps/actions/swap.ts]: @derHowie pushed!

Comment (derHowie) [reply src/raps/actions/swap.ts]: ahh good call, I got you half way there. haha.

**Review by derHowie — APPROVED** [review]

**Review by derHowie — APPROVED** [review]
👑

**Pull request merged**